shelley-ma/impl/src/Cardano/Ledger/Mary/Value.hs

```diff
@@ -154,26 +155,70 @@ instance CC.Crypto crypto => Val (Value crypto) where
   modifyCoin f (Value c m) = Value n m where (Coin n) = f (Coin c)
   pointwise p (Value c x) (Value d y) = (p c d) && (pointWise (pointWise p) x y)
 
-  size (Value _ v) =
-    -- add uint for the Coin portion in this size calculation
-    foldr accum uint v
+  {- Explanation of the Value size calculation :
+
+  The size calculation is to approximate the number of bytes in a
+  compact representation of Value (CompactValue). CompactValue has two constructors :
+
+  1. CompactValueAdaOnly is used when v == mempty
+  it takes a Word64 to represent an ada amount (unpacked in the compact representation)
+
+  2. CompactValueMultiAsset (used otherwise) takes an ada amount and token bundle data
+    i) Word64 (ada)
+    ii) Word (number of distinct types of multi-assets in the bundle)
+    iii) rep :
+      The rep consists of five parts
+        A) a sequence of Word64s representing quantities
+        B) a sequence of Word16s representing policyId indices
+        C) Word16s representing asset name indices
+           (as a special case for empty asset names,
+            the index points to the end of the string)
+        D) a blob of policyIDs
+        E) a blob of asset names
+  -}
+
```

Comment:
write units everywhere
Size in the heap of values, in words (to get the size in bytes multiply by 4 on a 32-bit machine or 8 on a 64-bit machine)

Word - 8 bytes

ignoring 32 bit machines

Reply:
ascii art of layout

decide on linking comments to/from the compact value definition